builders/interoperability/xcm/xc-registration/foreign-asset-registration.md

```diff
@@ -0,0 +1,134 @@
+---
+
+# Self-Serve Asset Registration for Sibling Parachains
+
+## Introduction {: #introduction }
+
+Registering your parachain’s native tokens on Moonbeam or Moonriver lets your community enjoy ERC‑20–style UX and deep EVM integrations while retaining full on‑chain provenance. This guide shows sibling Polkadot parachain teams how to self‑register a foreign asset using the new `ForeignAssetOwnerOrigin` introduced in Moonbeam Runtime 3600.
+
+## Why a New Origin? {: #why-a-new-origin }
+
+Moonbeam introduced a new dedicated origin called `ForeignAssetOwnerOrigin`, which only permits an XCM message whose origin contains the asset’s MultiLocation to execute calls in the `evm‑foreign‑assets` pallet. In practice, that means only the sovereign account of the parachain that owns the asset, or Moonbeam governance, can create, freeze, unfreeze, or relocate it. Alongside this, a configurable runtime constant called `ForeignAssetCreationDeposit` is reserved from the caller’s sovereign account at creation time. The deposit discourages spam registrations.
+
+## Required Deposits {: #required-deposits }
+
+To prevent spam, a `ForeignAssetCreationDeposit` is required and locked for the asset's lifetime. The deposit is funded from the sibling parachain's sovereign account on the Moonbeam network, which thus needs to be sufficiently funded to cover the asset deposit and the associated transaction fees. If the asset is destroyed through governance, the deposit is unreserved and returned to the original sovereign account.
+
+Deposits are network‑specific and can be adjusted by Moonbeam governance via the `parameters` pallet:
+
+=== "Moonbeam"
+
+     ```text
+     {{ networks.moonbeam.xcm.foreign_asset_deposit.display }} GLMR
+     ```
+
+=== "Moonriver"
+
+     ```text
+     {{ networks.moonriver.xcm.foreign_asset_deposit.display }} MOVR
+     ```
+
+=== "Moonbase Alpha"
+
+     ```text
+     {{ networks.moonbase.xcm.foreign_asset_deposit.display }} DEV
+     ```
+
+## Prerequisites {: #prerequisites }
+
+There are a few prerequisites to be aware of: 
+
+- The sibling parachain's [sovereign account](/builders/interoperability/xcm/core-concepts/sovereign-accounts/){target=\_blank} on Moonbeam must be sufficiently funded to cover the asset deposit and the transaction fees. It's recommended that you have an extra buffer of additional funds for any subsequent transactions. See this [guide to calculating a sovereign account](/builders/interoperability/xcm/core-concepts/sovereign-accounts/){target=\_blank}
+- Your parachain should support XCM V4
+
+## Assemble Your Asset Details {: #assemble-your-asset-details }
+
+Before registering your asset on Moonbeam, you'll need to formulate the following four parameters: asset ID, decimals, symbol, and name. Asset ID can be any integer between 0 and 255 that isn't already used as a foreign asset ID on Moonbeam. 
+
+```typescript
+const ASSET_ID    = 1;
```

Comment:
We recommend folks to calculate an Asset ID similar to before. Grab the multilocation in bytes, hash it, and slice to get the first 16 bytes of the hash, we reverse it and then get the hex representation of it. For the address we prepend with 0xFF... until 20 bytes (4 bytes which generally means 0xFFFFFFFF)